src/subdomains/lightning/services/lightning-transaction.service.ts

```diff
@@ -0,0 +1,278 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { LnBitsUsermanagerWalletDto } from 'src/integration/blockchain/lightning/dto/lnbits.dto';
+import { LndTransactionDto, LndTransactionResponseDto } from 'src/integration/blockchain/lightning/dto/lnd.dto';
+import { LightningHelper } from 'src/integration/blockchain/lightning/lightning-helper';
+import { LightningService } from 'src/integration/blockchain/lightning/services/lightning.service';
+import { LightingWalletRepository } from 'src/subdomains/user/application/repositories/lightning-wallet.repository';
+import { LightningClient } from '../../../integration/blockchain/lightning/lightning-client';
+import {
+  TransactionLightningEntity,
+  TransactionLightningState,
+  TransactionLightningType,
+} from '../entities/transaction-lightning.entity';
+import { TransactionOnchainEntity } from '../entities/transaction-onchain.entity';
+import { UserTransactionEntity, UserTransactionType } from '../entities/user-transaction.entity';
+import { TransactionLightningRepository } from '../repositories/transaction-lightning.repository';
+import { TransactionOnchainRepository } from '../repositories/transaction-onchain.repository';
+import { UserTransactionRepository } from '../repositories/user-transaction.repository';
+
+@Injectable()
+export class LightningTransactionService {
+  private readonly client: LightningClient;
+
+  constructor(
+    lightningService: LightningService,
+    private readonly transactionOnchainRepo: TransactionOnchainRepository,
+    private readonly transactionLightningRepo: TransactionLightningRepository,
+    private readonly userTransactionRepository: UserTransactionRepository,
+    private readonly lightingWalletRepository: LightingWalletRepository,
+  ) {
+    this.client = lightningService.getDefaultClient();
+  }
+
+  async syncOnchainTransactions(): Promise<TransactionOnchainEntity[]> {
+    const maxId = await this.transactionOnchainRepo.getMaxId();
+
+    if (!maxId) {
+      return this.saveOnchainTransactions(0);
+    }
+
+    const transactionEntity = await this.transactionOnchainRepo.findOneBy({ id: maxId });

```

Comment:
You can use the following code to get the latest transaction directly, and remove the `getMaxId` methods.
```ts
const lastTx = await this.transactionOnchainRepo.findOne({ where: {}, order: { block: 'DESC' } });
```

Reply:
Maybe add a helper method to BaseRepository?